cargo-apk/src/apk.rs

```diff
@@ -223,18 +223,49 @@ impl<'a> ApkBuilder<'a> {
             Profile::Custom(c) => c.as_str(),
         };
 
-        let signing_key = self.manifest.signing.get(profile_name);
-
-        let signing_key = match (signing_key, is_debug_profile) {
-            (Some(signing), _) => Key {
-                path: crate_path.join(&signing.path),
-                password: signing.keystore_password.clone(),
-            },
-            (None, true) => self.ndk.debug_key()?,
-            (None, false) => return Err(Error::MissingReleaseKey(profile_name.to_owned())),
+        let keystore_env = format!(
+            "CARGO_APK_{}_KEYSTORE",
+            profile_name.to_uppercase().replace('-', "_")
+        );
+        let password_env = format!("{}_PASSWORD", keystore_env);
+
+        let path = std::env::var_os(&keystore_env).map(PathBuf::from);
+        let password = std::env::var(&password_env).ok();
+
+        let signing_key = match (path, password) {
+            (Some(path), Some(password)) => Key { path, password },
+            (Some(path), None) if is_debug_profile => {
+                eprintln!(
+                    "{} not specified, falling back to default password",
+                    password_env
+                );
+                Key {
+                    path,
+                    password: ndk_build::ndk::DEFAULT_DEV_PASSWORD.to_owned(),
+                }
+            }
+            (_, _) => {
```

Comment:
`(None, _)` to be a bit more explicit?

Reply:
That makes it more complicated as there is no (Some, None) case then.

Reply:
That's actually a good catch, as it would be the perfect case for a warning/error explaining that if the user sets a  (non-dev) keystore path, they _must also_ provide the password for it - more reason to have it don't you think? :grin: